contracts/cw-proposal-single/src/tests.rs

```diff
@@ -387,489 +573,198 @@ fn do_test_votes(
-///
-/// This is a pretty nonsense passing threshold but helps demonstrate
-/// the issue well enough.
 #[test]
-fn test_close_votes_quorum() {
-    do_test_votes(
-        vec![
-            TestVote {
-                voter: "ekez".to_string(),
-                position: Vote::No,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::Yes,
-            },
-            TestVote {
-                voter: "keze".to_string(),
-                position: Vote::Yes,
-                weight: Uint128::new(5),
-                should_execute: ShouldExecute::Yes,
-            },
-            TestVote {
-                voter: "ezek".to_string(),
-                position: Vote::No,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::No,
-            },
-        ],
-        Threshold::ThresholdQuorum {
-            threshold: PercentageThreshold::Percent(Decimal::percent(10)),
-            quorum: PercentageThreshold::Majority {},
-        },
-        Status::Passed,
-        None,
-        None,
-    );
+fn test_votes_low_threshold() {
+    testing::test_votes_low_threshold(do_votes_cw20_balances);
+    testing::test_votes_low_threshold(do_votes_cw4_weights);
+    testing::test_votes_low_threshold(do_votes_staked_balances)
 }
 
 #[test]
 fn test_majority_vs_half() {
-    do_test_votes(
-        vec![
-            TestVote {
-                voter: "ekez".to_string(),
-                position: Vote::No,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::Yes,
-            },
-            TestVote {
-                voter: "keze".to_string(),
-                position: Vote::Yes,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::Yes,
-            },
-        ],
-        Threshold::ThresholdQuorum {
-            threshold: PercentageThreshold::Percent(Decimal::percent(50)),
-            quorum: PercentageThreshold::Majority {},
-        },
-        Status::Passed,
-        None,
-        None,
-    );
-
-    do_test_votes(
-        vec![
-            TestVote {
-                voter: "ekez".to_string(),
-                position: Vote::No,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::Yes,
-            },
-            TestVote {
-                voter: "keze".to_string(),
-                position: Vote::Yes,
-                weight: Uint128::new(10),
-                should_execute: ShouldExecute::No,
-            },
-        ],
-        Threshold::ThresholdQuorum {
-            threshold: PercentageThreshold::Majority {},
-            quorum: PercentageThreshold::Majority {},
-        },
-        Status::Rejected,
-        None,
-        None,
-    );
+    testing::test_majority_vs_half(do_votes_cw20_balances);
+    testing::test_majority_vs_half(do_votes_cw4_weights);
+    testing::test_votes_low_threshold(do_votes_staked_balances)
```

Comment:
copy paste error? shouldnt this be `test_majority_vs_half` also?

Reply:
Great catch! Fixed. Thanks.